leakcanary-android-instrumentation/src/main/java/leakcanary/InstrumentationLeakDetector.kt

```diff
@@ -143,14 +145,20 @@ class InstrumentationLeakDetector {
     val heapDumpUptimeMillis = SystemClock.uptimeMillis()
     KeyedWeakReference.heapDumpUptimeMillis = heapDumpUptimeMillis
 
+    val heapDumpDurationMillis: Long
+
     try {
-      Debug.dumpHprofData(heapDumpFile.absolutePath)
+      heapDumpDurationMillis = measureTimeMillis {
+        Debug.dumpHprofData(heapDumpFile.absolutePath)
+      }
     } catch (exception: Exception) {
       SharkLog.d(exception) { "Could not dump heap" }
       return AnalysisPerformed(
           HeapAnalysisFailure(
-              heapDumpFile, analysisDurationMillis = 0,
+              heapDumpFile = heapDumpFile,
               createdAtTimeMillis = System.currentTimeMillis(),
+              dumpDurationMillis = HeapAnalysis.DUMP_DURATION_UNKNOWN,
```

Comment:
Technically we could actually provide the duration here (ie start of heap dump to exception)

Reply:
I don't have a strong opinion, let me just calculate the duration